pkg/antctl/raw/check/installation/command.go

```diff
@@ -0,0 +1,394 @@
+	}
+
+	t.Log("Deploying echo-other-node Service...")
+	svc = newService(echoOtherNodeDeploymentName, map[string]string{"name": echoOtherNodeDeploymentName}, 80)
+	_, err = t.client.CoreV1().Services(t.namespace).Create(ctx, svc, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	echoOtherNodeDeployment := newDeployment(deploymentParameters{
+		Name:    echoOtherNodeDeploymentName,
+		Role:    kindEchoName,
+		Port:    80,
+		Image:   deploymentImage,
+		Command: []string{"/agnhost", "netexec", "--http-port=80"},
+		Affinity: &corev1.Affinity{
+			PodAntiAffinity: &corev1.PodAntiAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+					{
+						LabelSelector: &metav1.LabelSelector{
+							MatchExpressions: []metav1.LabelSelectorRequirement{
+								{Key: "name", Operator: metav1.LabelSelectorOpIn, Values: []string{clientDeploymentName}},
+							},
+						},
+						TopologyKey: "kubernetes.io/hostname",
+					},
+				},
+			},
+		},
+		Labels: map[string]string{"app": echoOtherNodeDeploymentName},
+	})
+	nodes, err := t.client.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to list Nodes: %s", err)
+	}
+	if len(nodes.Items) > 2 {
+		_, err = t.client.AppsV1().Deployments(t.namespace).Create(ctx, echoOtherNodeDeployment, metav1.CreateOptions{})
+		if err != nil {
+			return fmt.Errorf("unable to create Deployment %s: %s", echoOtherNodeDeploymentName, err)
+		}
+		if err := t.waitForDeploymentsReady(ctx, time.Second, podReadyTimeout, clientDeploymentName, echoSameNodeDeploymentName, echoOtherNodeDeploymentName); err != nil {
+			return err
+		}
+		t.echoOtherNodePod = map[string]string{}
+		echoOtherNodePod, err := t.client.CoreV1().Pods(t.namespace).List(ctx, metav1.ListOptions{LabelSelector: "name=" + echoOtherNodeDeploymentName})
+		if err != nil {
+			return fmt.Errorf("unable to list Echo Other Node Pod: %s", err)
+		}
+		for _, echoOtherNodePod := range echoOtherNodePod.Items {
+			t.echoOtherNodePod[echoOtherNodePod.Name] = echoOtherNodePod.Status.PodIP
+		}
+	} else {
+		t.Log("Skipping other node Deployments as multiple nodes are not available")
+		if err := t.waitForDeploymentsReady(ctx, time.Second, podReadyTimeout, clientDeploymentName, echoSameNodeDeploymentName); err != nil {
+			return err
+		}
+	}
+	t.clientPods, err = t.client.CoreV1().Pods(t.namespace).List(ctx, metav1.ListOptions{LabelSelector: "kind=" + kindClientName})
+	if err != nil {
+		return fmt.Errorf("unable to list Client Pods: %s", err)
+	}
+	t.echoSameNodePod = map[string]string{}
+	echoSameNodePod, err := t.client.CoreV1().Pods(t.namespace).List(ctx, metav1.ListOptions{LabelSelector: "name=" + echoSameNodeDeploymentName})
+	if err != nil {
+		return fmt.Errorf("unable to list Echo same node Pod: %s", err)
+	}
+	for _, echoSameNodePod := range echoSameNodePod.Items {
+		t.echoSameNodePod[echoSameNodePod.Name] = echoSameNodePod.Status.PodIP
+	}
```

Comment:
this assumes a single IP address per Pod, we should use `.Status.PodIPs` and support the dual stack case